src/main/java/org/broadinstitute/hellbender/tools/exome/hmm/CopyNumberTriStateHiddenMarkovModel.java

```diff
@@ -0,0 +1,354 @@
+     * @throws IllegalArgumentException if any of the inputs (target or state) is {@code null}.
+     */
+    @Override
+    public double logTransitionProbability(final CopyNumberTriState currentState, final Target fromTarget,
+                                           final CopyNumberTriState nextState, final Target nextPosition) {
+        Utils.nonNull(currentState);
+        Utils.nonNull(nextState);
+        Utils.nonNull(fromTarget);
+        Utils.nonNull(nextPosition);
+        if (currentState == CopyNumberTriState.NEUTRAL) {
+            // inter-event or event start transition prob.:
+            return nextState == CopyNumberTriState.NEUTRAL ? log1Minus2P : logP;
+        } else {
+            final double distance = calculateDistance(fromTarget, nextPosition);
+            final double logF = -distance * invD;
+            final double log1MinusF = Math.log1p(-Math.exp(logF));
+            if (nextState == CopyNumberTriState.NEUTRAL) {
+                // End of event transition prob.:
+                return MathUtils.logSumLog(new double[] {logF + logQ, log1MinusF + log1Minus2P});
+
+                //return MathUtils.logSumLog(new double[] {logF + logQ, log1MinusF + log1Minus2P});
+            } else if (currentState == nextState) {
+                // Event extension transition prob.:
+                return MathUtils.logSumLog(new double[] {logF + log1MinusQ, log1MinusF + logP});
+            } else {
+                // Event "sign switch" (del. <--> dup.) transition prob.:
+                return log1MinusF + logP;
+            }
+        }
+    }
+
+    /**
+     * Calculate the distance between two targets.
+     * <p>
+     * If the target provided do not contain intervals, the distance is set to 0 in order to revert
+     * to the non-distance dependent model described in Fromer et al. paper
+     * <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3484655/table/tbl1/">Table 1</a>.
+     * </p>
+     * <p>
+     * If both targets map to different chromosomes then we return {@link Double#POSITIVE_INFINITY}.
+     * </p>
+     * @param fromTarget the previous target.
+     * @param toTarget the next target.
+     * @return any values between 0 and {@link Double#POSITIVE_INFINITY}.
+     * @throws NullPointerException if any of the targets is {@code null}.
+     */
+    @VisibleForTesting
+    double calculateDistance(final Target fromTarget, final Target toTarget) {
+        final SimpleInterval fromInterval = fromTarget.getInterval();
+        final SimpleInterval toInterval = toTarget.getInterval();
+        if (fromInterval == null || toInterval == null) {
+            return 0.0;
+        } else if (!fromInterval.getContig().equals(toInterval.getContig())) {
+            return Double.POSITIVE_INFINITY;
+        } else {
+            final double distance = toInterval.getStart() - fromInterval.getEnd() - 1;
+            return distance > 0 ? distance : Math.max(0, fromInterval.getStart() - toInterval.getEnd() - 1);
```

Comment:
@vruano I'm not sure what the right behavior is, but I feel like the distance between overlapping targets should never be *more* than the distance between non-overlapping targets.  I think returning `0` would make sense, as would an exception, as would redefining distance to be the distance between the starts.  I truly have no preference and leave it up to you.